packages/wrangler/src/pages.tsx

```diff
@@ -1013,7 +1017,7 @@ const createDeployment: CommandModule<
     let builtFunctions: string | undefined = undefined;
     const functionsDirectory = join(cwd(), "functions");
     if (existsSync(functionsDirectory)) {
-      const outfile = join(tmpdir(), "./functionsWorker.js");
+      const outfile = join(tmpdir(), `./functionsWorker-${Math.random()}.js`);
```

Comment:
@GregBrimble - these changes are due to the `tmpdir()` function always returning the same folder, which resulted in parallel tests overwriting each other's files.

We should come up with a better solution than actually needing to write these files to disk, ideally.

Also we could consider using `fs.mkdtempSync()` in combination with `tmpdir()` to really encapsulate separate runs of Pages builds.

But for now I think this is a reasonable way to unblock this PR.